src/Chunk.cpp

```diff
@@ -1145,20 +1145,33 @@ int cChunk::GrowCactus(int a_RelX, int a_RelY, int a_RelZ, int a_NumBlocks)
 			for (auto & Coord : Coords)
 			{
 				if (
-					UnboundedRelGetBlockType(a_RelX + Coord.x, Top + 1, a_RelZ + Coord.z, BlockType) &&
+					UnboundedRelGetBlockType(a_RelX + Coord.x, Top + i, a_RelZ + Coord.z, BlockType) &&
 					(
 						cBlockInfo::IsSolid(BlockType) ||
 						(BlockType == E_BLOCK_LAVA) ||
 						(BlockType == E_BLOCK_STATIONARY_LAVA)
 					)
 				)
 				{
+					// Remove cactus
+					GetWorld()->DigBlock(a_RelX + GetPosX() * cChunkDef::Width, Top + i, a_RelZ + GetPosZ() * cChunkDef::Width);
```

Comment:
Doesn't `DigBlock` already create the pickups?

Reply:
It doesn't. From the [API](https://api.cuberite.org/cWorld.html#DigBlock_1): `Replaces the specified block with air, without dropping the usual pickups for the block.`